pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1505,10 +1507,19 @@ public void deleteOfflineTable(String tableName) {
     LOGGER.info("Deleting table {}: Removed table config", offlineTableName);
 
     // Remove instance partitions
+    final String rawTableName = TableNameBuilder.extractRawTableName(tableName);
     InstancePartitionsUtils.removeInstancePartitions(_propertyStore,
-        InstancePartitionsType.OFFLINE.getInstancePartitionsName(TableNameBuilder.extractRawTableName(tableName)));
+        InstancePartitionsType.OFFLINE.getInstancePartitionsName(rawTableName));
     LOGGER.info("Deleting table {}: Removed instance partitions", offlineTableName);
 
+    // Remove table locker if there
+    if (_tableUpdaterLockMap.containsKey(rawTableName)) {
```

Comment:
The key is `offlineTableName` instead of `rawTableName`.
Also, no need to do check and remove here, directly remove should be good enough
```suggestion
    _tableUpdaterLockMap.remove(offlineTableName);
```